### ISSUE TITLE
reportInvalidTypeVarUse reporting in overloads

### ISSUE THREAD

**Issue by chdsbd** [microsoft/pyright#1894]
**Is your feature request related to a problem? Please describe.**

If an overloaded function uses a TypeVar multiple times, but one overload only uses the TypeVar once, reportInvalidTypeVarUse reports an issue.

```py
from typing import overload, TypeVar, Union, Dict, Any, List
from typing_extensions import Literal

class Document: ...

_T = TypeVar("_T", bound="Union[Document, Dict[str, Any]]")
@overload
def create(
    doc_or_docs: _T, # reportInvalidTypeVarUse flags this 
    load: Literal[False],
) -> int: ...
@overload
def create(
    doc_or_docs: List[_T],
    load: Literal[True] = ...,
) -> List[_T]: ...
@overload
def create(
    doc_or_docs: List[_T],
    load: Literal[False],
) -> List[int]: ...
@overload
def create(
    doc_or_docs: _T,
    load: Literal[True] = ...,
) -> _T: ...

```

**Describe the solution you'd like**

If an overloaded function uses a TypeVar multiple times, but one overload only uses the TypeVar once, reportInvalidTypeVarUse should not report an issue. The workaround is to copy the TypeVar bound into the function definition, but that doesn't feel great. It seems like this code should be allowed.

**Comment by bschnurr** [comment]
Not sure if this is correct but maybe you can split the first overload up instead of using `_T`
```
@overload
def create(
    doc_or_docs: Document, 
    load: Literal[False],
) -> int: ...


@overload
def create(
    doc_or_docs: Dict[str, Any], 
    load: Literal[False],
) -> int: ...
```

**Comment by erictraut** [comment]
The TypeVar use is invalid for that overload signature, so I think this error is still valid.

In my opinion, the correct solution is to copy the TypeVar bound into the signature. That's what you would need to do in the non-overload case. 

If you don't like this behavior, you can suppress the error using `# type: ignore` or by disabling the `reportInvalidTypeVarUse` check for that file using `# pyright: reportInvalidTypeVarUse=false`.

**Comment by chdsbd** [comment]
makes sense, thanks for the quick response!
